### PR TITLE
Make citeproc commonjs-loadable

### PR DESCRIPTION
**Pull request by retorquere**

### CODE REVIEW

**Comment by fbennett** [comment]
Looks like a good move. Will merge tomorrow after double-checking.

**Comment by retorquere** [comment]
Whoops, I forgot you will also need a package JSON. I edited this directly on github so I don't know how to add new changes to this pull request, but you'll need something like this:

```
{
  "name": "citeproc",
  "description": "A JavaScript implementation of the Citation Style Language (CSL)",
  "author": "Frank Bennett (https://citeproc-js.readthedocs.io/)",
  "version": "1.1.157",
  "main": "citeproc.js"
}
```

and then follow [these instructions](http://evanhahn.com/make-an-npm-baby/) on how to publish to npm.

This will make a serious difference to BBT -- I'm looking to simplify a whole lot of code as I try my umpteenth attempt at a 5.0 port.

**Comment by retorquere** [comment]
come to think of it, is `citeproc.js` the result of a compilation of the files in `src`? If so, adding the `module.exports` in `citeproc.js` would be undone on the next compilation.

**Comment by fbennett** [comment]
Yes, it will need tweaks to the homespun Python test/build script. I'll look at that. Is the idea that you want the processor set up as an npm module, and to have it updated in npm on every release?

**Comment by retorquere** [comment]
That would be very helpful for me.

Going npm may have other benefits BTW; npm has [version bumping built-in](https://docs.npmjs.com/cli/version), and I'm in the process of converting my javascript-assembly-chain to webpack which in my case is a lot faster than the ruby scripts I had. That's a side benefit really to the conceptual cleanup that came with it, which I'll need for the 5.0 port.

**Comment by dsifford** [comment]
@fbennett I can help publish on npm if you need it. There's a few important lesser known things that should be done when publishing; one of the most important of which is including an `.npmignore` file to reduce bloat.

**Comment by fbennett** [comment]
(This may not get done this weekend. I'll be putting in some time on replacing the disk in my laptop.)

**Comment by retorquere** [comment]
No worries. @dsifford will know a lot more about this than the one-liner I threw in. I'm on a patched version anyhow while #35 is open.

**Comment by fbennett** [comment]
Thanks for offers of help with npm. I've worked with it before. I won't be replacing the current processor build script, which works just fine. The aim will just be to incorporate an npm release to the release-time choreography.

File: citeproc.js [hunk 1]
```diff
@@ -16737,3 +16737,5 @@ CSL.parseParticles = function(){
         }
     }
 }();
+
+if (typeof module != 'undefined) { module.exports = CSL }
```
Comment (retorquere) [inline citeproc.js]: Whoops, sorry about that.

**Comment by fbennett** [comment]
Closing this. The `commonjs`-compatible module has been published to `npm` as `citeproc`.

**Comment by dsifford** [comment]
@fbennett Is the citeproc package kept up to date? If not, can I send a PR to this repo so that everything is contained here?

**Comment by fbennett** [comment]
Ach. This slipped from notice. So there are two npm packages that I need to update on releases. Will bump it later.

**Comment by dsifford** [comment]
@fbennett Why not just create a package.json and `dist` folder in this repo and save yourself some work? I can get that set up for you if you'd be interested.

**Comment by fbennett** [comment]
Derek,

That would be very welcome, if you have time.

FB

**Pull request closed without merging**